subprojects/testing-base/src/main/java/org/gradle/api/tasks/testing/AbstractTestTask.java

```diff
@@ -321,7 +346,7 @@ public void onOutput(Closure closure) {
      * @param closure The closure to call.
      */
     public void beforeSuite(Closure closure) {
-        testListenerBroadcaster.add(new ClosureBackedMethodInvocationDispatch("beforeSuite", closure));
+        testListenerSubscriptions.addListener(new ClosureBackedMethodInvocationDispatch("beforeSuite", closure));
```

Comment:
🎨 Extract common expression to method: `addListenerSubscription("beforeSuite", closure)`

Reply:
I had that at some point, took it out to reduce the diff to the bare minimum. Will add back.

Reply:
Done.